Swope brandi install updated (wtg on developer) (#6968)
open-cluster-management/rhacm-docs: install/uninstall.adoc

[#uninstalling]
= Uninstalling

When you uninstall {acm}, you see two different levels of the uninstall process: A _custom resource removal_ and a _complete operator uninstall_. The uninstall process can take up to 20 minutes and the process includes removing resources.

> - The custom resource removal is the first and most basic type of uninstall that removes the custom resource of the `MultiClusterHub` instance, but leaves other required operator resources. This level of uninstall is helpful if you plan to reinstall with the same settings and components.

Author: formatting only

- The complete operator uninstall is the second level process that removes most operator components, excluding components such as custom resource definitions. When you continue with this step, it removes all of the components and subscriptions that were not removed with the custom resource removal. After this uninstall, you must reinstall the operator before reinstalling the custom resource.

[#prerequisite-detach]
== Prerequisites

> * If you have managed clusters attached, you need to detach them. *Note:* This does not include the `local-cluster`, which is your self-managed hub cluster. For more information about detaching clusters, see the _Removing a cluster from management_ section in link:../clusters/cluster_lifecycle/create_intro.adoc#create-intro[Cluster creation introduction]. 

Author: @cameronmwall this is where we removed the errors and instead wrote it to prevent errors.

Reviewer: this looks good to me

> * If you use Discovery, you need to disable the function. From the console, go to the `Discovered Clusters` table and click *Disable cluster discovery*. Confirm that you want to remove the service. You can also use the terminal to disable Discovery with the following command:

Reviewer 2: ```suggestion
* If you use Discovery, you need to disable the function. From the console, go to the _Discovered Clusters_ table and click *Disable cluster discovery*. Confirm that you want to remove the service. You can also use the terminal to disable Discovery with the following command:
```

+
[source,bash]
----
oc delete discoveryconfigs --all --all-namespaces
----

> * If you use Agent Service Configuration, disable and remove the `AgentServiceConfig` resource.Complete the following steps:

Reviewer 2: ```suggestion
* If you use Agent Service Configuration, disable and remove the `AgentServiceConfig` resource. Complete the following steps:
```

.. Select the *Options* menu for the `MultiClusterObservability` custom resource. 

.. Select *Delete MultiClusterObservability*. 

+
When you delete the resource, the pods in the `open-cluster-management-observability` namespace on {acm-short} hub cluster, and the pods in `open-cluster-management-addon-observability` namespace on all managed clusters are removed.

[#removing-a-multiclusterhub-instance-by-using-commands]

> == Removing _multiclusterhub_ resources by using commands

Reviewer 2: ```suggestion
== Removing _MultiClusterHub_ resources by using commands
```

> Delete the `MultiClusterHub` custom resource and remove artifacts.

Author: added this description @cameronmwall

Reviewer: I think this is helpful

> Delete the `MultiClusterHub` custom resource and remove artifacts.

Reviewer 2: ```suggestion
Delete the `MultiClusterHub` custom resource and remove artifacts. Complete the following steps:
```

+

> .. Run the script to remove any possible artifacts that remain from the previous installation. If there are no remaining artifacts, a message is returned that no resources were found.

Reviewer 2: I recall we didn't want to have scripts in the doc, but we can revisit this in a refresh

Author: Yes, there is a whole other issue for this and the main reason for starting to clean up this file. You are right, though I think we have to have it in this case, it just should be part of another process and not embedded here. Also, it is in a troubleshooting file. 

This is a result of just patching up the doc as told, without understanding or revisiting the big picture. So when I get all this in, the next issue takes care of this. You are right.

> Next you can uninstall the`MultiClusterHub` operator. *Note:* If you plan to reinstall the same {product-title-short} version, you do not need to uninstall the operator.

Reviewer 2: Should this be listed as a step instead of saying "Next"?
```suggestion
. Uninstall the`MultiClusterHub` operator. *Note:* If you plan to reinstall the same {acm-short} version, you do not need to uninstall the operator.
```

Author: This content changes if I can ever get this PR merged because of the rewrites for uninstall. I will leave for now bc it will move and change. 

To answer, ideally we **don't** use **next** because it is implied. Here it is used (I think) because though the overall task is to uninstall, you have to do a number of other things first, which is why the file is complicated. 

It can be used as a transition; for instance if you have a whole section of preparing to install, then the next section is actually install and you need to transition to the next section or header: _Next, complete the install in the following procedure._ 

I have to rewrite a lot of this, so I didn't mess too much with this particular area, the whole script is going away.

[#deleting-the-components-by-using-the-console]
== Deleting the components by using the console

> When you use the {ocp} console to uninstall, you remove the `MultiClusterHub` resource to delete the object. Wait for the status, then delete the operator. Complete the following steps to uninstall by using the console:

Author: @cameronmwall can you check this line 145?

Reviewer: I think we may want to adjust the phrasing here. I think we should emphasize deleting the MCH to remove the object and then say uninstall the operator. This might be personal preference on my part but seems more intuitive

Author: changed.
